tangelo/linq/simulator_base.py

```diff
@@ -284,6 +398,54 @@ def _get_expectation_value_from_statevector(self, qubit_operator, state_prep_cir
 
         return expectation_value
 
+    def _get_variance_from_statevector(self, qubit_operator, state_prep_circuit, initial_statevector=None):
+        r"""Take as input a qubit operator H and a state preparation returning a
+        ket |\psi>. Return the variance <\psi | H | \psi>, computed
+        without drawing samples (statevector only). Users should not be calling
+        this function directly, please call "get_expectation_value" instead.
+
+        Args:
+            qubit_operator(openfermion-style QubitOperator class): a qubit
+                operator.
+            state_prep_circuit: an abstract circuit used for state preparation
+                (only pure states).
+
+        Returns:
+            complex: The variance of this operator with regards to the
+                state preparation.
+        """
+        n_qubits = state_prep_circuit.width
+
+        variance = 0.
+        prepared_frequencies, prepared_state = self.simulate(state_prep_circuit, return_statevector=True, initial_statevector=initial_statevector)
+
+        # Otherwise, use generic statevector expectation value
+        for term, coef in qubit_operator.terms.items():
+
+            if len(term) > n_qubits:  # Cannot have a qubit index beyond circuit size
+                raise ValueError(f"Size of operator {qubit_operator} beyond circuit width ({n_qubits} qubits)")
+            elif not term:  # Empty term: no simulation needed
+                variance += 0.
+                continue
+
+            if not self.n_shots:
```

Comment:
I don't think this is true. It's fine for use in computing standard error, but variance can still be non-zero with statevector. I'm not sure the best way to compute variance here in the exact case. Do Var(H) = <H^2> - <H>^2 directly?

Reply:
Can something like https://github.com/goodchemistryco/Tangelo/blob/7a32ef2e72f24fd393b010e0a8500da4afae6188/tangelo/linq/simulator_base.py#L347
could be used in that situation? I mean just converting the statevector to ideal frequencies, and then resuse `_get_variance_from_frequencies` to get the ideal variance.

Reply:
Looks like we didn't need extra logic! Everything is computed the same way with frequencies now, either ideal or re-sampled depending on n_shots.